tools/c7n_azure/c7n_azure/resources/cdn_custom_domain.py

```diff
@@ -58,3 +58,48 @@ def extra_args(cls, parent_resource):
                 'endpoint_name': endpoint_name,
                 'profile_name': profile_name
             }
+
+
+@CdnCustomDomain.filter_registry.register('tls-version')
+class CdnManagedHttpsParametersFilter(ValueFilter):
+    """Find all Cdn Endpoint Custom Domain with minimum Tls Version set to TLS12
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: cdn-endpoint-custom-domain-tls-version
+            resource: azure.cdn-custom-domain
+            filters:
+               - and:
+                  - type: value
+                    key: properties.customHttpsProvisioningState
+                    op: eq
+                    value: Enabled
+                  - type: tls-version
+                    key: properties.customHttpsParameters.minimumTlsVersion
+                    op: eq
+                    value: "TLS12"
+
+    """
+    schema = type_schema(
+        'tls-version',
+        rinherit=ValueFilter.schema,
+        value = {'type': 'string', 'enum':['None', 'TLS10', 'TLS12']})
+
+    def process(self, resources, event=None):
+        for custom_domain in resources:
+            properties = custom_domain.get('properties', {})
+            if 'customHttpsParameters' not in properties:
+                http_params = {
```

Comment:
can this not be done with existing policy value filters, like

```
or:
  - customHttpsParameter: absent
  - type: value
     key: customHttpsPresent.minimumTlsVersion: 
     op: not-equal
     value: TLS12
```


its also unclear where these static values are coming from, it would need a doc link, afaict these are not the default server side values, but I didn't see where to find them.

Reply:
@kapilt I doubt if the existing value filters can work but I will test it. Let me share the link with you for more reference: https://learn.microsoft.com/en-us/python/api/azure-mgmt-cdn/azure.mgmt.cdn.models.cdnmanagedhttpsparameters?view=azure-python